azure-pipelines.yml

```diff
@@ -65,6 +65,16 @@ jobs:
   - script: bundle exec bin/test/cucumber.rb
     displayName: "exec cucumber"
 
+  # azureStorage* env vars are defined in the pipeline UI as secret variables
+  - bash: "./bin/ci/az-publish.sh"
+    env:
+      AZURE_STORAGE_ACCOUNT: $(azureStorageAccount)
+      AZURE_STORAGE_KEY: $(azureStorageKey)
+      AZURE_STORAGE_CONNECTION_STRING: $(azureStorageConnectionString)
+      SOURCE_BRANCH: $(Build.SourceBranch)
+    displayName: "Publish to Azure Blob Storage"
+    condition: and(succeeded(), eq(variables['Agent.JobName'], variables['mainJob']))
```

Comment:
Can we change this to explicitly branch on the Xcode version?

Like this maybe?

```
eq(variables['XCODE_VERSION'], '10.2.1')
```

Reply:
Sure, done.
@jmoody what do you think, should we explicitly add the macOS version check in condition too?

Reply:
The condition should use the Xcode version explicitly.

Reply:
Honestly, I prefer `mainJob` approach because we will need to change only 1 place when the default configuration is updated (for example, when Xcode 10.3 is released)

With this approach we will have clear separation: `main job` and `additional jobs`.
Main job runs all tasks including publishing to the Azure Blob, run `appcenter-cli` and etc.
Additional jobs run only testing.

Reply:
I agree.  I missed the addition of the "main job" variable.  This is a good idea.

I was worried the "main job" was an azure pipeline concept meaning "the first job in the matrix" or something.

Reply:
Returned condition with `mainJob`